### PR TITLE
drivers: mbox: mbox_esp32: fix mbox callback call

### PR DESCRIPTION
**Pull request by uLipe**
Fixes the mailbox notification callback that was not being called from the ISR when the other side sends something.

### CODE REVIEW

**Comment by uLipe** [comment]
Not sure why this was merged this way, fortunately my local copy had the correct code.

**Comment by sylvioalves** [comment]
Would you mind running c-land-format in this file to fix the macro spaces?

File: drivers/mbox/mbox_esp32.c [hunk 1]
```diff
@@ -73,20 +74,22 @@ IRAM_ATTR static void esp32_mbox_isr(const struct device *dev)
 		;
 
 	if (dev_data->cb) {
+		/* For ESP32 soft mbox driver, the messag parameter of the callback holds
```
Comment (raffarost) [inline drivers/mbox/mbox_esp32.c:77]: nit: messag -> message

**Review by raffarost — DISMISSED** [review]

**Comment by uLipe** [comment]
@sylvioalves just run the clang-format, PTAL again.

**Review by raffarost — APPROVED** [review]
I just noticed this driver lacks taking IRQ priority and flags from DTS.
I'll submit a PR after yours gets merged

**Comment by uLipe** [comment]
@raffarost  i think when this driver got merged, the priority properties and flags were not present in the upstream yet.

**Review by sylvioalves — APPROVED** [review]

**Review by LucasTambor — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Pull request merged**